### PR TITLE
Support for edit action on Policy

### PR DESCRIPTION
**Pull request by GaneshSPatil**
##### Description:

* Add an ability to define edit action as part of the policy.
* Allow users to define edit action policy on roles SPA.
* Modify Internal Environments API v1 to return user permissions.
* Modify Internal Config Repos API to return user permissions.
* Fix environments SPA to enable/disable action buttons based on permissions.
* Fix config repo SPA to enable/disable action buttons based on permissions.

### CODE REVIEW

**Comment by GaneshSPatil** [comment]
Things to verify are documented at: https://github.com/gocd/gocd/issues/7390#issuecomment-564388931

File: server/src/main/webapp/WEB-INF/rails/webpack/models/new-environments/environments.ts [hunk 1]
```diff
@@ -142,7 +161,7 @@ export class EnvironmentWithOrigin extends ValidatableMixin {
 
   clone(): EnvironmentWithOrigin {
     return new EnvironmentWithOrigin(this.name(),
-                                     true,
+                                     new Permissions(true, true),
```
Comment (kritika-singh3) [inline server/src/main/webapp/WEB-INF/rails/webpack/models/new-environments/environments.ts]: Why not retain the original value or set it to default `false`?

**Comment by kritika-singh3** [comment]
Create environment modal - always shows the spinner : 

Screen Shot 2019-12-11 at 1 55 18 PM

**Comment by GaneshSPatil** [comment]
These changes have been reverted as part of https://github.com/gocd/gocd/pull/7435 PR.

**Pull request merged**